### DISCUSSION TITLE
Adding components to the Login page.

### DISCUSSION THREAD

**Discussion by chris-diam** [keycloakify/keycloakify#146] [Q&A]
Hi @garronej !!

I am working on a project that uses keycloak for user authentication, I have managed to customize the CSS of the login page but I am struggling to understand how I can add some custom components, like an eye icon in the password textField in order to toggle on/off the visibility of the password etc. 
Could you provide some advice for this issue?
Thanks a lot.

**Comment by garronej** [comment]
Hi @chris-diam,  

You can start from [this repo](https://github.com/garronej/keycloakify-demo-app/tree/look_and_feel) and hack around to understand how things works.  

For the eye, on [this app](https://github.com/garronej/keycloakify-demo-app/tree/look_and_feel) we have it.  

<img width="603" alt="image" src="https://user-images.githubusercontent.com/6702424/181255435-b67a737c-06df-4fe3-9d87-c13843e1ca79.png">


And [here](https://github.com/InseeFrLab/onyxia-web/blob/ffe3288942f6d9b2ba7c9d2eae23c61b673f008f/src/ui/components/KcApp/Login/Login.tsx#L199-L211) is the code.  

Good luck.

**Reply by chmcquee** [reply]
@chris-diam  Did you find a solution? I have the same question

**Reply by chmcquee** [reply]
@garronej it would be nice to also show the solution in a thread like this, as it's not clear to me from the code you used.

**Reply by garronej** [reply]
I mean, it's kind of out of scope for keycloakify.  
Keycloakify enables you to build your register page with React and integrate with Keycloak.  
Building the actual page is up to you, that's the point.  
I suggest you to have a look at https://mui.com/, there is a ton a great components that you can use directly in your login pages.  

Also the starter is now this: https://github.com/garronej/keycloakify-advanced-starter

**Reply by chmcquee** [reply]
Okay cool, yeah I found something good [here](https://stackoverflow.com/questions/60391113/how-to-view-password-from-material-ui-textfield) thanks for pointing me in the right direction :)

**Reply by garronej** [reply]
@chmcquee your welcome 😊   
The SO snipet is kind of outdated though.  
There you have the reference code that you can use as is:  
<img width="909" alt="image" src="https://user-images.githubusercontent.com/6702424/216102411-a50a12ae-ec82-4742-8961-e899576d6af5.png">

https://mui.com/material-ui/react-text-field/#input-adornments

**Reply by chris-diam** [reply]
Hello again! I have another question, I know this may be stupid but I am struggling when it comes to customizing the login-reset-password and generally how to change the messages e.g. the "Forgot Password" header as depicted below.
Thanks in advance!
![image](https://user-images.githubusercontent.com/91747939/217531296-ba922776-2675-46f9-a91c-7fc6c42e3651.png)

**Reply by garronej** [reply]
Please delete this message and open another discution,  
Discutions and issues are resources that can help others

**Comment by chris-diam** [comment]
Thank you for your response! 

I am currently developing the "my-extra-page-1" component but I am facing difficulties when it comes to adding functionality to the "Sign In" Button. Is it possible to make it functional without using the Template component?

Thanks in advance.

**Reply by garronej** [reply]
Be aware V6 of Keycloakify is just around the corner.
